libpldmresponder/fru.cpp

```diff
@@ -22,118 +26,117 @@ namespace responder
 
 constexpr auto root = "/xyz/openbmc_project/inventory/";
 
-std::optional<pldm_entity>
-    FruImpl::getEntityByObjectPath(const dbus::InterfaceMap& intfMaps)
+pldm_entity FruImpl::getEntityByObjectPath(const dbus::ObjectValueTree& objects,
+                                           const std::string& path)
 {
-    for (const auto& intfMap : intfMaps)
+    pldm_entity entity{};
+
+    auto iter = objects.find(path);
+    if (iter == objects.end())
+    {
+        return entity;
+    }
+
+    for (const auto& intfMap : iter->second)
```

Comment:
replace auto& intfMap  with auto intfMap if there are no intension to modify original value because for loop don't update anyof the original value

Reply:
I think we need to keep the reference for this variable because even though we are not updating the value we need to make sure we take the updated interface map at runtime when getEntityByObjectPath is called IMO